### PR TITLE
feat: add management lock

### PR DESCRIPTION
**Pull request by Saelmala**
## This PR adds:
- A lock on production management (e.g. add sources to production, delete sources from production, start production)
- A lock on managing all productions (e.g. deleting productions)
- A lock on editing sources

To unlock the actions, simply click on the lock. The aim of this is to prevent accidental removals etc. The default is therefore that the actions are locked. 

## Images:

Edit production, locked:

Edit production, unlocked:

Delete production, locked:

Delete production, unlocked:

Source edit view, locked:

Source edit view, unlocked:

Running production, locked:

Running production, unlocked:

### CODE REVIEW

**Comment by Saelmala** [comment]
> In the UI examples - it looks like the lock on a production also disables the possibility to create a new production?

Yes, Linda and I discussed this yesterday - will change it

File: src/app/page.tsx [hunk 1]
```diff
@@ -22,13 +22,7 @@ function Home() {
...
           </div>
         </div>
-      </div>
-      <CreateProduction />
-      <div className="flex items-center w-full">
-        <Suspense fallback={<LoadingCover />}>
-          {/* @ts-expect-error Async Server Component: https://github.com/vercel/next.js/issues/42292 */}
-          <ProductionsList />
-        </Suspense>
+        <HomePageContent productions={productions} />
```
Comment (LucasMaupin) [inline src/app/page.tsx:25]: Maybe we can fetch the productions directly in the HomePageContent component.

**Comment by Saelmala** [comment]
> > In the UI examples - it looks like the lock on a production also disables the possibility to create a new production?
> 
> Yes, Linda and I discussed this yesterday - will change it

Fixed.

**Review by LucasMaupin — APPROVED** [review]

**Comment by Saelmala** [comment]
Made it possible to add new production and monitor in locked mode:

image

**Review by ABLindberg — APPROVED** [review]
Looks good to me. Others to review the code.

**Pull request merged**